common/src/main/java/com/microsoft/identity/common/internal/result/MsalBrokerResultAdapter.java

```diff
@@ -176,6 +177,29 @@ public static BrokerResult brokerResultFromBundle(@NonNull final Bundle resultBu
                 );
     }
 
+    public static boolean getHelloResultFromBundle(final Bundle bundle) throws ClientException {
+        final String methodName = ":getHelloResultFromBundle";
+        if (bundle == null) {
+            Logger.warn(TAG + methodName, "The hello result bundle is null.");
+            return false;
+        } else if (!StringUtil.isEmpty(bundle.getString(AuthenticationConstants.Broker.NEGOTIATED_BP_VERSION_KEY))) {
+            final String negotiatedBrokerProtocolVersion = bundle.getString(AuthenticationConstants.Broker.NEGOTIATED_BP_VERSION_KEY);
+            Logger.verbose(TAG + methodName,
+                    "Able to establish the connect, " +
+                            "the broker protocol version in common is ["
+                            + negotiatedBrokerProtocolVersion + "]");
+            return true;
+        } else if (bundle.get(AuthenticationConstants.Broker.BROKER_RESULT_V2) != null
+                && bundle.get(AuthenticationConstants.Broker.BROKER_RESULT_V2) instanceof BrokerResult) {
+            //no common broker version found and create the client exception from the result bundle.
```

Comment:
Add comments to say we are using `serializable` for back compat

Reply:
As discussed, do not use `BrokerResult` to send result here, instead use `errorcode` and `errormessage` as is and for back compat, have this fall back logic